helper/schema/core_schema.go

```diff
@@ -172,6 +172,53 @@ func (r *Resource) CoreConfigSchema() *configschema.Block {
 		}
 	}
 
+	// insert configured timeout values into the schema
+	if r.Timeouts != nil {
```

Comment:
Ahh, I like that it's possible to leave this out of the schema in the common case where it's not used. I was worried initially that we'd have a redundant extra `timeouts` thing in _every_ schema!